themes/agnoster.zsh-theme

```diff
@@ -109,7 +109,9 @@ prompt_git() {
   if $(git rev-parse --is-inside-work-tree >/dev/null 2>&1); then
     repo_path=$(git rev-parse --git-dir 2>/dev/null)
     dirty=$(parse_git_dirty)
-    ref=$(git symbolic-ref HEAD 2> /dev/null) || ref="➦ $(git rev-parse --short HEAD 2> /dev/null)"
+    ref=$(git symbolic-ref HEAD 2> /dev/null) || \
+    ref=" $(git describe --exact-match HEAD 2> /dev/null)" ||\
```

Comment:
I had the same solution in mind, but for me I had to add `--tags` to the `git describe` command. Otherwise it searches only for annotated tags, and fails on normal tags.

Reply:
Also, theoretically the `formats` and `actionformats` can be used to get `ref` more cleanly. Also, if you checked out a commit that is pointed to by both a branch and a tag it will choose what to display based on how you checked it out.

The downside is that it needs to be formatted conditionally anyways (depending if it's a commit hash, branch or tag) so I'm not sure it's worth it.